### PR TITLE
Initial commit of the Magento 2 Optimize Flat Table module

### PR DESCRIPTION
**Pull request by rbnmulder**

### CODE REVIEW

**Review by royduin — APPROVED** [review]

**Comment by royduin** [comment]
Any suggestions @indykoning or @jbclaudio? Will be merged after your approval.

File: Plugin/Magento/Eav/Model/Entity/AttributePlugin.php [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        }
+        return $result;
+    }
+
+    public function isEnabled(): bool
+    {
+        return $this->scopeConfig->isSetFlag('justbetter/flat_tables/enabled');
+    }
+
+    public function getMargin($type = 'varchar'): int
```
Comment (claudio-ferraro) [inline Plugin/Magento/Eav/Model/Entity/AttributePlugin.php]:
```suggestion
    public function getMargin(string $type = 'varchar'): int
```

File: Plugin/Magento/Eav/Model/Entity/AttributePlugin.php [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+use Zend_Db_Expr as Zend_Db_Expr;
+
+class AttributePlugin
+{
+    public function __construct(
+        protected ScopeConfigInterface $scopeConfig,
+        protected Attribute            $attributeResource
+    )
+    {
+    }
```
Comment (claudio-ferraro) [inline Plugin/Magento/Eav/Model/Entity/AttributePlugin.php]:
```suggestion
    public function __construct(
        protected ScopeConfigInterface $scopeConfig,
        protected Attribute $attributeResource
    ) {
    }
```

File: Plugin/Magento/Eav/Model/Entity/AttributePlugin.php [hunk 3]
```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace JustBetter\OptimizeFlatTables\Plugin\Magento\Eav\Model\Entity;
+
+use Magento\Catalog\Model\ResourceModel\Attribute;
+use Magento\Eav\Model\Entity\Attribute\AbstractAttribute;
+use Magento\Framework\App\Config\ScopeConfigInterface;
+use Zend_Db_Expr as Zend_Db_Expr;
```
Comment (claudio-ferraro) [inline Plugin/Magento/Eav/Model/Entity/AttributePlugin.php]: The alias is the same as the namespace. Is an alias even necessary then?
```suggestion
use Zend_Db_Expr;
```

File: etc/module.xml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0"?>
+<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:framework:Module/etc/module.xsd">
+    <module name="JustBetter_OptimizeFlatTables" setup_version="1.0.0">
+        <sequence>
+            <module name="Magento_Eav" />
```
Comment (claudio-ferraro) [inline etc/module.xml:5]: I see that `Magento\Catalog\Model\ResourceModel\Attribute` is also used. So wouldn't it be better to add `Magento_Catalog` here as well?
```suggestion
            <module name="Magento_Eav" />
            <module name="Magento_Catalog" />
```

**Review by claudio-ferraro — APPROVED** [review]
🚀

**Pull request merged**